app/services/dttp/retrieve_contact.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Dttp
+  class RetrieveContact
+    include ServicePattern
+
+    class HttpError < StandardError; end
+
+    attr_reader :trainee
+
+    CONTACT_ATTRIBUTE_FILTERS = %w[
+      emailaddress1
+      firstname
+      lastname
+    ].freeze
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      response = Client.get("/contacts(#{trainee.dttp_id})?$select=#{filters}")
+      if response.code != 200
+        raise HttpError, "status: #{response.code}, body: #{response.body}, headers: #{response.headers}"
+      end
+
+      JSON(response.body)
```

Comment:
Do you think we should create a class to wrap these to abstract the DTTP JSON schema away from our app? Maybe `Dttp::Contact` that takes the JSON and then has properly named accessors.

Reply:
Sure, do you mean to switch the DTTP accessors (emailaddress1, firstname etc) to Trainee column names (email_address, first_name etc?)

Reply:
Yes, I think we could try that. Just something that will abstract away all of the inconsistencies in the JSON naming would be good.